xbmc/addons/AddonRepos.cpp

```diff
@@ -0,0 +1,129 @@
+/*
+ *  Copyright (C) 2005-2020 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "AddonRepos.h"
+
+#include "Addon.h"
+#include "AddonSystemSettings.h"
+#include "CompileInfo.h"
+#include "utils/StringUtils.h"
+#include "utils/log.h"
+
+#include <vector>
+
+using namespace ADDON;
+
+/**********************************************************
+ * CAddonRepos
+ *
+ */
+
+CAddonRepos::CAddonRepos()
+{
+  SetupOfficialRepoInfos();
+}
+
+void CAddonRepos::SetupOfficialRepoInfos()
+{
+  // get our official repo tuples into vector
+  const std::vector<std::string> officialAddonRepos =
+      StringUtils::Split(CCompileInfo::GetOfficialAddonRepos(), ',');
+
+  RepoInfo newRepoInfo;
+  for (const auto& addonRepo : officialAddonRepos)
+  {
+    const std::vector<std::string> tmpRepoInfo = StringUtils::Split(addonRepo, '|');
+    newRepoInfo.m_repoId = tmpRepoInfo.front();
+    newRepoInfo.m_origin = tmpRepoInfo.back();
+    m_officialRepoInfos.emplace_back(newRepoInfo);
+  }
+}
+
+const std::vector<RepoInfo>& CAddonRepos::GetOfficialRepoInfos() const
+{
+  return m_officialRepoInfos;
+}
+
+bool CAddonRepos::IsFromOfficialRepo(const AddonPtr& addon, bool bCheckAddonPath) const
+{
+  auto comparator = [&](const RepoInfo& officialRepo) {
+    if (bCheckAddonPath)
+    {
+      return (addon->Origin() == officialRepo.m_repoId &&
+              StringUtils::StartsWithNoCase(addon->Path(), officialRepo.m_origin));
+    }
+
+    return addon->Origin() == officialRepo.m_repoId;
+  };
+
+  return addon->Origin() == ORIGIN_SYSTEM ||
+         std::any_of(GetOfficialRepoInfos().begin(), GetOfficialRepoInfos().end(), comparator);
+}
+
+const std::map<std::string, AddonPtr>& CAddonRepos::GetLatestPrivateVersions() const
+{
+  return m_latestPrivateVersions;
+}
+
+const std::map<std::string, AddonPtr>& CAddonRepos::GetLatestOfficialVersions() const
+{
+  return m_latestOfficialVersions;
+}
+
+void CAddonRepos::SetupAddonMaps(const VECADDONS& addons)
+{
+  m_addonsByRepoMap.clear();
+
+  for (const auto& addon : addons)
```

Comment:
Maybe here instead:

```cpp
for (const auto& addon : addons)
{
  if (addon.IsCompatible())
    m_addonsByRepoMap[addon->Origin()].insert({addon->ID(), addon});
}
```

Probably better here as there is no point in ever loading addons that can't be compatible.